Move default_url_options set up to an initializer

sauloperez opened the pull request:
### What? Why?

For some reason running `bundle exec rake db:migrate RAILS_ENV=staging` on a deployment fails with:

```
rake aborted!
NameError: uninitialized constant Spree::Config
```

Running `bundle exec rails server` for instance, does not. There must be a difference on the way a rake task and the rails commands load the app.

Moving this configuration to an initializer, at the end of the initialization process, fixes it. The constant `Spree::Config` is already loaded.

**This is preventing the release v1.22.0 from being staged and tested**

### What should we test?

This should fix the deployment to staging therefore, I can test it myself.

sauloperez commented:
thanks for the idea of trying out an initializer @Matt-Yorkley !

Matt-Yorkley commented:
![teamwork](https://media.giphy.com/media/wrzf9P70YWLJK/giphy.gif)

Matt-Yorkley commented on config/initializers/action_mailer.rb:
```diff
@@ -0,0 +1,4 @@
+ActionMailer::Base.configure do |config|
+  # Use https when creating links in emails
+  config.default_url_options = { protocol: 'https', host: Spree::Config[:site_url] }
```
Ah, I think we actually want to only do this in staging and production. Test and development have very different settings for links, emails and the site url.

sauloperez replied:
yes, that's right. We don't want to mess with that.

daniellemoorhead commented:
@sauloperez Pls, #assignyourself 😱

daniellemoorhead commented:
Also, how is this to be tested? Does it have to be tested? There's no detail in the description.

sauloperez commented:
sorry, sorry :trollface:

sauloperez commented:
> Also, how is this to be tested? Does it have to be tested? There's no detail in the description.

Added it. Thanks for the heads-up. I gave for granted the scope of this PR was understood.

sauloperez commented:
Fixed!

```
TASK [deploy : migrate database] ***********************************************
Wednesday 31 October 2018  09:09:47 +0100 (0:00:00.305)       0:01:20.374 *****
changed: [staging.katuma.org]
```
